### ISSUE TITLE
Notes added in Simple note android sometimes do not appear in SimpleNote on Linux

### ISSUE THREAD

**Issue by bro1** [Automattic/simplenote-electron#354]
#### Steps to reproduce
1. Set up and log on in Android app and Linux app synchronize both.
2. Add notes in simple note on android (probably while not connected to network)
3. Launch SimpleNote Linux client and let it sync
4. Launch Android app while connected to network and let it sync
#### What I expected

The notes that were added on Android to show up on Linux app
#### What happened instead

The notes that were added on Android app do not appear on Linux app.
However the data was not lost:
1. it's still in Android app
2. it can also be accessed via web interface

Additionally, notes that are added at a later point while android device is connected to the network, synchronise between apps fine.

**Comment by roundhill** [comment]
Hmm, works for me. Are you using version 1.0.2 of the Linux app?

**Comment by bro1** [comment]
Hi,

I tried to follow those steps and no, it's not reproducible with the steps that I have described in the original issue report.

However, at this stage I still have a very similar issue:
* Online application access
* 2 Android devices (Both of them showing everything as up to date)
* SimpleNote 1.0.2 on Linux (has all the notes)
* SimpleNote 1.0.2 on Windows 7 (with strange behavior):
   a) it does not any notes that I have added over the last few days through other clients 
   b) it has all the tags that I have added in other clients
   c) it still has tags that I have removed using other clients
   d) I have added a new note (Note A) through Windows client today and the note ended up in all other apps (Android and Linux)
   e) Today I have modified Note A in Linux app
   f) Those changes did not show up in Windows app :(


Is there any technical information that I could extract that would help to investigate this?

**Comment by bro1** [comment]
More development on this issue:

Linux client got out of sync as well:
 * the notes that existed in Linux client still keep synchronising 
 * new notes added through Linux client show up online and in Android
 * any new notes added through other clients do not show up in Linux

**Comment by chainedghost** [comment]
I have the same issue on Ubuntu 16.04.
The web app and android stay in sync. The desktop app doesn't pull the changes made on the other apps, but it does send changes.
I end up with older versions of the note on all devices whenever I open the desktop app.

**Comment by bro1** [comment]
I have jus upgraded my Linux client to 1.0.3 (+ log-out + logged in ). This resynchronised everything. I'll check if that makes any difference in the synchronisation behaviour.

**Comment by chainedghost** [comment]
A quick check after upgrading to 1.0.3 and it looks like the issue is resolved.

**Comment by chainedghost** [comment]
Nevermind, after the first successful sync, the issue is back again.

**Comment by roundhill** [comment]
I'm still unable to reproduce this, unfortunately. Are your windows/linux devices using a proxy at all? Any other details you can provide, especially clear repro steps would be helpful!

**Comment by jacklenox** [comment]
This affects me too. Using Elementary OS (based on Ubuntu 16.04), with 1.0.3.

**Comment by bro1** [comment]
It was all working fine for a few weeks after upgrading to 1.0.3 on Linux Mint 18. 
However the issue reappeared today again:
* I added a note on my Android Simplenote at around lunchtime
* Launched Simplenote on Linux machine several hours later and the note did not appear in there

Can the developers provide any info how to capture any useful diagnostics? Logs/command line flags, etc.

**Comment by zorgeus** [comment]
I have the same issue.
Sync is fine for Android and iOS, but Win7 client doesn't show notes that was added from Android.
Solution: Logout/login on win7 client.

**Comment by roundhill** [comment]
> Can the developers provide any info how to capture any useful diagnostics? Logs/command line flags, etc.

Here's a build that will show the developer tools, if you don't mind running it with it open. When you notice the issue, have a look at the console window and let us know if you see anything interesting in there:

https://cloudup.com/files/iGgHkFjgdzK/download

**Comment by bro1** [comment]
OK, I downloaded Windows version and there does not seem to be any useful information displayed in the console. 

And I still have the same issue: 
1. added a note in Android yesterday.
2. started Windows application today (the version from https://cloudup.com/files/iGgHkFjgdzK/download)
3. the note that I have added yesterday is not there
4. added a test note in Windows it appeared everywhere it showed up in android as well
5. added a second test note in Android it showed up in Windows too. But no changes in the console output. See attachment below:

![capture](https://cloud.githubusercontent.com/assets/397284/18901939/94ce0a4c-85ab-11e6-9c5c-8060a2a0e5cf.PNG)

**Comment by dmsnell** [comment]
@bro1 thanks for the helpful information!

:question: for you or anyone - is syncing broken in this way 100% of the time or is it sporadic…sometimes you write a note and it appears on your Android device, other times it doesn't…

**Comment by bro1** [comment]
Hi, it's sporadic.

I am not sure if it's just me finding false patterns, but the two scenarios that I think I may have noticed that precede the missing notes in Desktop based apps are these events:
* adding notes on Android by sharing note from another app to Simple note
* adding notes on Android phone while on mobile connection

**Comment by roundhill** [comment]
The `Connected` debug line does tell is a lot. That means that the websocket connection was established successfully, so I think we can rule out connectivity and/or authorization.

**Comment by chainedghost** [comment]
It seems to have something to do with the timeline. You should be able to reproduce with the following:
1. Create note on any device, except the desktop app.
2. Editing the note between a web browser or android app keeps edits in sync so that the last edit is seen on the next device.
3. Open the note on the desktop app and it is up to date. Make edits.
4.Opening the not from android or the web browser will show the edits made on the desktop app. Make edits.
5. Open the note on the desktop app. The edits made on the other devices are gone. If you just open the note on other devices at this point, your previous edits are also gone. Everything defaults to your original desktop app edits. At this point I also noticed that if I take the timeline back a tick, the latest edits made on the other devices show up.

In other words the desktop app defaults back to the last edits made on the desktop and not on other devices, but those edits are available in the timeline.

**Comment by dmsnell** [comment]
@chainedghost this awesome sleuthing is so incredibly valuable to us in finding the cause. I think that what you discovered about having the external edits positioned on the sidebar is critical here - we must be loading the note out of indexedDB and then immediately syncing with Simperium before Simperium syncs with us (my guess is that the "initial state of the note" is a blank data structure since all IndexedDB operations are async)

**Comment by noamelf** [comment]
I also encounter this issue :disappointed:

**Comment by roundhill** [comment]
We've fixed up a few issues related to sync, here's a beta of the next release: https://github.com/Automattic/simplenote-electron/releases/tag/untagged-6a81612f542e4dcb0391. Let us know how it works for you.

**Comment by bro1** [comment]
Thanks... I have been using this beta (Linux) version for 8 days and have not had any synch issues.

**Comment by Harrri** [comment]
Hi evrybody,
I'm now using 1.0.5 and still have the problem.
The only way, to get notes created on android to linux is to delete ~/.config/Simplenote/IndexedDB and restart simplenote.
My proposal: there must be a subroutine in your code, which builds up the local database in ~/.config/Simplenote/IndexedDB when it's missing. Why not add a "sync"-button which does just call that subroutine?
Thanks for reading
Harrri

**Comment by chrisfromthelc** [comment]
A user is reporting the same issue with the 1.0.5 version on Linux (Fedora 24 & Arch Linux).

> I have created few new notes on my phone (android app) and I can see this notes on your server but linux app (on two computers - fedora 24 & arch linux) does not show new notes but I can see new tags that are defined in this new notes... When I create new note in linux app it is immediately on your server and I see this new note immediately also on android app and on another computer with linux app. It seems to me that linux app has trouble to sync new notes that are created through android app?

**Comment by mankurt** [comment]
Have a similar issue. Notes created or deleted on android & mac not updating on linux. Using simplenote 1.0.5 on archlinux. Signing out of the simplenote app and signing back in re-syncs everything correctly.

**Comment by zeke** [comment]
Has anyone been able to reproduce these issues on the macOS Electron Desktop client? I am going to give it a shot.

@dmsnell, @roundhill any pro-tips on how to enable more verbose dev debugging in the Electron app?

**Comment by roundhill** [comment]
> @dmsnell, @roundhill any pro-tips on how to enable more verbose dev debugging in the Electron app?

Yup, you can enable them by uncommenting this line: https://github.com/Automattic/simplenote-electron/blob/master/desktop/app.js#L50

**Comment by roundhill** [comment]
I think [this comment](https://github.com/Automattic/simplenote-electron/issues/354#issuecomment-250077557) is the biggest clue here. He was able to create notes from the electron app, and they sync fine. But the issue is that it does not appear to be requesting the changes from simperium when it reconnects in some cases.

**Comment by dmsnell** [comment]
Hi all. I'm going to close this issue and merge it into #385 for organizational purposes. I'm pretty sure that it's the same core issue causing the sync problems in all of the apps build off of this repository (Windows, Linux).

We'll be tracking there and you are free/welcome to participate in that discussion!

**Comment by marcorighi** [comment]
Year 2018. Android Client do not sync notes with Linux and website data. It remains in a not sync status and it is not possible sync it!

**Comment by luqiang21** [comment]
@marcorighi Same here, I edited inside website on Mac. The new notes are not synced, I tried to edit and change networks. None has been working. Don't know how to sync these notes...

**Comment by sporto** [comment]
Same for me, notes that I created in the Android app do not appear in the Desktop or Web app. The other way around is fine.

**Comment by theoneandonly-vector** [comment]
it happens again..

**Comment by johuhr** [comment]
but this time there's no sync at all between any devices. 
This is the case since about 1 week now. 
Looks like the server is down...

**Comment by johuhr** [comment]
ups - I updated the app on android. Now everything just works fine again!

**Comment by theoneandonly-vector** [comment]
Yeah same here :)
